src/main/groovy/com/bmuschko/gradle/docker/DockerRegistryCredentials.groovy

```diff
@@ -15,10 +15,13 @@
  */
 package com.bmuschko.gradle.docker
 
+import groovy.transform.CompileStatic
+import org.gradle.api.credentials.PasswordCredentials
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Optional
 
-class DockerRegistryCredentials {
+@CompileStatic
+class DockerRegistryCredentials implements PasswordCredentials {
```

Comment:
Thanks for providing the pull request. I think it would make sense to add a test case that shows the problem it solves. Is it correct to assume that it is required for using the plugin from the Kotlin DSL? I recently broke out functional test fixtures for Kotlin-based builds so adding a test case shouldn't be too hard.